src/main/java/seedu/address/model/Scheme/SchemeRetrieval.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.scheme;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import seedu.address.model.person.Person;
+
+/**
+ * Handles the logic of retriving schemes for a person.
+ */
+public class SchemeRetrieval {
+
+    private final Person targetFamily;
+    private final double income;
+    private final int age;
+    private final int familySize;
+
+    private final int incomePerCapita;
+
+    private final ArrayList<Scheme> schemes = new ArrayList<>();
+
+    private final ArrayList<Scheme> allSchemes = new ArrayList<>();
+
+    /**
+     * Constructor for SchemeRetrieval.
+     */
+    public SchemeRetrieval(Person targetFamily) {
+        this.targetFamily = targetFamily;
+        this.income = targetFamily.getIncome().getValue();
+        LocalDate currentDate = LocalDate.now();
+        this.age = targetFamily.getDateOfBirth().toLocalDate().getYear() - currentDate.getYear();
```

Comment:
This way of calculating the age does not take into account the person's date of birth?
Better to use Java's `Period.between(LocalDate, LocalDate).toYears()` instead.